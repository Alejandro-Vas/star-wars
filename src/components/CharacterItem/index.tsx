import Badge from 'components/Badge';
import Property from 'components/Property';

import styles from './styles.module.scss';
import { ICharacterEntity } from '../../interfaces/index';

interface ICharacterItem {
  character:ICharacterEntity
}

function CharacterItem({ character }:ICharacterItem) {
  const {
    name,
    height,
    mass,
    birth_year: birthYear,
    gender,
  } = character;

  return (
    <div className={styles.root}>
      <div className={styles.title}>
        {name}
      </div>

      <div className={styles.wrapper}>
        <Property amount={height} type="height" />

        <Property amount={mass} type="mass" />
      </div>

      <div className={styles.wrapper}>
        <Badge text={gender} color="green" />

        <Badge text={birthYear} color="green" />
      </div>
    </div>
  );
}

export default CharacterItem;
