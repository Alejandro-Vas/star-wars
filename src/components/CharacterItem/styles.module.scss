@import "styles/variables.scss";

.root {
  padding: 5px 28px 4px 28px;
  max-width: 352px;
  background: $paper;
  border-radius: 8px;
  filter: drop-shadow(4px 4px 4px rgba(33, 33, 33, 0.1));
  cursor: pointer;
}

.title {
  margin-bottom: 12px;
  font-weight: 700;
  font-size: 18px;
  line-height: 21px;
  color: $textLight;
  text-shadow: 4px 4px 4px rgba(33, 33, 33, 0.1);
}

.wrapper {
  display: flex;
  gap: 13px;
  margin-bottom: 11px;
}

