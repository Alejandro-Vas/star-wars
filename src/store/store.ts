import { configureStore } from '@reduxjs/toolkit';
import { swApi } from './api/swApi';
import filterSlice from './slices/filter';
import translationSlice from './slices/translation';

export const store = configureStore({
  reducer: {
    [swApi.reducerPath]: swApi.reducer,
    filter: filterSlice.reducer,
    translation: translationSlice.reducer,
  },
  middleware: (getDefaultMiddleware) => getDefaultMiddleware()
    .concat(swApi.middleware),
});

export type TypeRootState = ReturnType<typeof store.getState>;
