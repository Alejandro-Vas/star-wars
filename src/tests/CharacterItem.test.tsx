import { render, screen } from '@testing-library/react';
import React from 'react';
import { Provider } from 'react-redux';
import { store } from 'store/store';
import CharacterItem from '../components/CharacterItem';

import lukeSkywalkerMock from './__mocks__/charactersMocks';

global.React = React;

const characterItem = render(
  <Provider store={store}>
    <CharacterItem character={lukeSkywalkerMock} />
  </Provider>,
);

describe('CharacterItem', () => {
  test('CharacterItem snapshot', () => {
    expect(characterItem).toMatchSnapshot();
  });

  test('CharacterItem snapshot', () => {
    expect(characterItem).toMatchSnapshot();
  });
});

export {};
