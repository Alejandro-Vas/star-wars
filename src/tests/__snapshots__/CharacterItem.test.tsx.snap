// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CharacterItem CharacterItem snapshot 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="root"
      >
        <div
          class="title"
        >
          Luke Skywalker
        </div>
        <div
          class="wrapper"
        >
          <div
            class="root"
          >
            <div
              class="amount"
            >
              172
            </div>
            <div
              class="type"
            >
              height
            </div>
          </div>
          <div
            class="root"
          >
            <div
              class="amount"
            >
              77
            </div>
            <div
              class="type"
            >
              mass
            </div>
          </div>
        </div>
        <div
          class="wrapper"
        >
          <div
            class="root male"
          >
            male
          </div>
          <div
            class="root "
          >
            19BBY
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="root"
    >
      <div
        class="title"
      >
        Luke Skywalker
      </div>
      <div
        class="wrapper"
      >
        <div
          class="root"
        >
          <div
            class="amount"
          >
            172
          </div>
          <div
            class="type"
          >
            height
          </div>
        </div>
        <div
          class="root"
        >
          <div
            class="amount"
          >
            77
          </div>
          <div
            class="type"
          >
            mass
          </div>
        </div>
      </div>
      <div
        class="wrapper"
      >
        <div
          class="root male"
        >
          male
        </div>
        <div
          class="root "
        >
          19BBY
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`CharacterItem CharacterItem snapshot 2`] = `
{
  "asFragment": [Function],
  "baseElement": <body />,
  "container": <div />,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
